test/integration/vaultdynamicsecret_integration_test.go

```diff
@@ -336,10 +334,37 @@ func TestVaultDynamicSecret(t *testing.T) {
 					assertDynamicSecretRotation(t, ctx, crdClient, &vdsObjFinal)
 				})
 			}
-
 			assert.Greater(t, count, 0, "no tests were run")
 		})
 	}
+	// Delete remaining CRDs which were created, and then validate that the leases are all revoked.
+	for _, c := range created {
+		// test that the custom resources can be deleted before tf destroy
+		// removes the k8s namespace
+		assert.Nil(t, crdClient.Delete(ctx, c))
+	}
+	// Get a Vault client so we can validate that all leases have been removed.
+	cfg := api.DefaultConfig()
+	cfg.Address = vaultAddr
+	c, err := api.NewClient(cfg)
+	assert.NoError(t, err)
+	c.SetToken(vaultToken)
+	// Check to be sure all leases have been revoked.
+	retry.DoWithRetry(t, "waitForAllLeasesToBeRevoked", 30, time.Second, func() (string, error) {
+		// ensure that all leases have been revoked.
+		resp, err := c.Logical().ListWithContext(ctx, fmt.Sprintf("sys/leases/lookup/%s/creds/readonly", outputs.DBRole))
```

Comment:
Remind me how this works? I was thinking it would be more like `sys/leases/lookup/<outputs.DBPath>/creds/<output.DBRole>`.

Reply:
Ah, I grabbed this from https://developer.hashicorp.com/vault/docs/commands/lease#examples
I think you're right on that path, I'll look into it.
I have validated in the controller logs though that the leases are revoked, but this does look off thx!

Reply:
Yup that was it, thanks! Did confirm that I can see the # of leases getting smaller through that loop on shutdown!